plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -2099,14 +2103,25 @@ public ConnectorTableProperties getTableProperties(ConnectorSession session, Con
                     bucketing.getColumns().stream()
                             .map(ColumnHandle.class::cast)
                             .collect(toImmutableList())));
+
+            if (isOptimizeSortedFileReads(session)) {
+                Map<String, ColumnHandle> columnHandles = getColumnHandles(session, table);
+                sortingProperties = hiveTable.getBucketHandle().map(HiveBucketHandle::getSortedBy)
+                        .orElse(ImmutableList.of())
+                        .stream()
+                        .map(sortingColumn -> new SortingProperty<>(
+                                columnHandles.get(sortingColumn.getColumnName()),
+                                sortingColumn.getOrder().getSortOrder()))
+                        .collect(toImmutableList());
+            }
         }
 
         return new ConnectorTableProperties(
                 predicate,
                 tablePartitioning,
                 Optional.empty(),
                 discretePredicates,
-                ImmutableList.of());
+                sortingProperties);
```

Comment:
Correctness question -- users are known to do bucket evolution over time.
So, before we can say the data is sorted, shouldn't we check the a partition's bucketing attributes, for each partition the query is going to read?

Reply:
Yes, though a bucket count or bucketing columns change wouldn't affect local sorting property (we just need the file to be in sorted order), it is possible that the sorting definition has changed order time.

Reply:
I've now added a check in HiveSplitManager to detect if sorting at partition level would not satisfy the sorting criteria defined at table level and throw an exception if `optimize_sorted_file_reads` is true.

Reply:
This is good as long as the flag is off by default. 

Can you please add a test with such a table?

- when flag is not set, the table should be read succesfully
- when optimization is enabled, the table read should fail
- when optimization is enabled, the table read should succeed, when partition pruning excludes the problematic partition

Reply:
Added the above tests now